video/out/opengl/shader_cache.c

```diff
@@ -831,6 +831,7 @@ static void gl_sc_generate(struct gl_shader_cache *sc,
 
     bstr *hash_total = &sc->tmp[5];
 
+    ADD(hash_total, "backend %s\n", sc->ra->name);
```

Comment:
What for?

Reply:
The commit message explains it

Reply:
Not really. You don't reuse the cache across backends.

Reply:
> You don't reuse the cache across backends.

`--opengl-shader-cache-dir` loads the shader from the filesystem, indexed only by the `hash_total`. If two different backends result in the same GLSL, the shader_cache may load the file contents of one and use it as the input cached_program for the other, resulting in UB.

Reply:
Loading invalid cached shaders shouldn't be UB, or mpv would trigger UB just when the user changes or updates GL drivers.

Reply:
I think loading a vulkan pipeline cache that doesn't have the vulkan pipeline cache header might be UB. (But I guess I could make ra_vk check for header presence). Vulkan pipeline caches have a spec-defined header format, not implementation-defined. (This header exists **for** the purpose of letting implementations check for compatibility)

I could test it on nvidia I guess to see what happens when you load garbage data with vkCreatePipelineCache.

Reply:
ra_gl actually adds its own "header" to the shader binary, because the API requires carrying along a format identifier.

Reply:
The format identifiers are shared between vulkan and opengl